### PR TITLE
Empty sets

### PR DESCRIPTION
**Pull request by axxroytovu**
Add code to build specific sets that do include decks, boosters, and sealed product but no cards

### CODE REVIEW

File: mtgjson5/set_builder.py [hunk 1]
```diff
@@ -404,7 +404,13 @@ def build_mtgjson_set(set_code: str) -> Optional[MtgjsonSetObject]:
     # Ensure we have a header for this set
     set_data = get_scryfall_set_data(set_code)
     if not set_data:
-        return None
+        with constants.RESOURCE_PATH.joinpath("additional_sets.json").open(
+            encoding="utf-8"
+        ) as f:
+            additional_sets_data = json.load(f)
+            set_data = additional_sets_data.get(set_code.lower(), False)
```
Comment (ZeldaZach) [inline mtgjson5/set_builder.py]: No need to add a False, as the default of `get(key)` is None. Also, I like using set_code UPPER if possible across the code base

File: mtgjson5/__main__.py [hunk 2]
```diff
@@ -102,6 +103,13 @@ def dispatcher(args: argparse.Namespace) -> None:
         return
 
     sets_to_build = ScryfallProvider().get_sets_to_build(args)
+    if args.all_sets:
+        with constants.RESOURCE_PATH.joinpath("additional_sets.json").open(
```
Comment (ZeldaZach) [inline mtgjson5/__main__.py]: This code is used in two places, can we abstract it out?

File: mtgjson5/__main__.py [hunk 3]
```diff
@@ -102,6 +103,13 @@ def dispatcher(args: argparse.Namespace) -> None:
...
+            encoding="utf-8"
+        ) as f:
+            additional_set_keys = set(json.load(f).keys())
+            additional_set_keys -= set(args.skip_sets)
+            sets_to_build += list(additional_set_keys)
```
Comment (ZeldaZach) [inline mtgjson5/__main__.py]: I'm not sure if loading, subtracting, and adding is correct as there could be duplicates.

File: mtgjson5/providers/cardkingdom.py [hunk 4]
```diff
@@ -178,11 +175,15 @@ def update_sealed_product(
 
             sealed_product.identifiers.card_kingdom_id = str(product["id"])
 
-            sealed_product.category = sealed_product.determine_mtgjson_sealed_product_category(
-                sealed_product.name.lower()
+            sealed_product.category = (
```
Comment (ZeldaZach) [inline mtgjson5/providers/cardkingdom.py]: Did tox clean this file up here? Feels unmodified

File: mtgjson5/providers/tcgplayer.py [hunk 5]
```diff
@@ -333,11 +333,15 @@ def generate_mtgjson_sealed_product_objects(
             sealed_product.identifiers.tcgplayer_product_id = str(product["productId"])
             sealed_product.release_date = product["presaleInfo"].get("releasedOn")
 
-            sealed_product.category = sealed_product.determine_mtgjson_sealed_product_category(
-                sealed_product.name.lower()
+            sealed_product.category = (
```
Comment (ZeldaZach) [inline mtgjson5/providers/tcgplayer.py]: This file should probably be left as it was, unless there's a change I can't tell

File: mtgjson5/resources/additional_sets.json [hunk 6]
```diff
@@ -0,0 +1,70 @@
+{
+    "dd3": {
```
Comment (ZeldaZach) [inline mtgjson5/resources/additional_sets.json]: CAPS please

File: mtgjson5/resources/additional_sets.json [hunk 7]
```diff
@@ -0,0 +1,70 @@
...
+        "digital":false,
+        "nonfoil_only":false,
+        "foil_only":false,
+        "search_uri":"https://api.scryfall.com/cards/search?include_extras=true\u0026include_variations=true\u0026order=set\u0026q=e%3Aq04\u0026unique=prints"
+    },
+    "q05": {
+        "name":"Challenger Decks 2021",
+        "code":"q05",
+        "set_type":"box",
+        "icon_svg_uri":"https://svgs.scryfall.io/sets/default.svg?1688356800",
```
Comment (ZeldaZach) [inline mtgjson5/resources/additional_sets.json]: I wouldn't include a cache buster ID, just the ".svg" should be enough

File: mtgjson5/set_builder.py (hunk 1 above)
Comment (axxroytovu) [reply mtgjson5/set_builder.py]: Fixed

File: mtgjson5/__main__.py (hunk 2 above)
Comment (axxroytovu) [reply mtgjson5/__main__.py]: I thought about building a provider but we don't use providers for any other resource file.

File: mtgjson5/__main__.py (hunk 3 above)
Comment (axxroytovu) [reply mtgjson5/__main__.py]: Fixed

File: mtgjson5/providers/cardkingdom.py (hunk 4 above)
Comment (axxroytovu) [reply mtgjson5/providers/cardkingdom.py]: Yeah this was a tox change and I didn't question it. I can remove these if you want.

File: mtgjson5/resources/additional_sets.json (hunk 6 above)
Comment (axxroytovu) [reply mtgjson5/resources/additional_sets.json]: Fixed

File: mtgjson5/resources/additional_sets.json (hunk 7 above)
Comment (axxroytovu) [reply mtgjson5/resources/additional_sets.json]: Ok. The scryfall api provides the cache ID so I just kept it.

File: mtgjson5/__main__.py (hunk 2 above)
Comment (ZeldaZach) [reply mtgjson5/__main__.py]: I don't think we need a provider, more so a common util method to get access to this data

File: mtgjson5/set_builder.py [hunk 8]
```diff
@@ -404,7 +404,13 @@ def build_mtgjson_set(set_code: str) -> Optional[MtgjsonSetObject]:
     # Ensure we have a header for this set
     set_data = get_scryfall_set_data(set_code)
     if not set_data:
-        return None
+        with constants.RESOURCE_PATH.joinpath("additional_sets.json").open(
+            encoding="utf-8"
+        ) as f:
+            additional_sets_data = json.load(f)
+            set_data = additional_sets_data.get(set_code.upper(), False)
```
Comment (ZeldaZach) [inline mtgjson5/set_builder.py]:
```suggestion
            set_data = additional_sets_data.get(set_code.upper())
```

File: mtgjson5/set_builder.py [hunk 9]
```diff
@@ -404,7 +404,13 @@ def build_mtgjson_set(set_code: str) -> Optional[MtgjsonSetObject]:
...
+        if not set_data:
```
Comment (ZeldaZach) [inline mtgjson5/set_builder.py:409]: I don't see a need for this if statement tbh, as set_data is either set or it's None

Comment (axxroytovu) [reply mtgjson5/set_builder.py:409]: If set_data is None, then it needs to close the function and not try building the rest of the data with no inputs. Line 406 checks if the set code was valid in Scryfall, and line 412 checks if the set code is valid in the local file. If the set code is in neither repository, it returns None.

File: mtgjson5/__main__.py (hunk 2 above)
Comment (axxroytovu) [reply mtgjson5/__main__.py]: Reworked to use util function

**Pull request merged**